### DISCUSSION TITLE
Compact title bar mode under adwaita

### DISCUSSION THREAD

**Discussion by liquidev** [ghostty-org/ghostty#3784] [Feature Requests, Ideas] [answered]
I'm just trying out ghostty, and I really like it so far! One thing I'd like to see though is the ability to change the huge header bar into a more compact, but still GTK-themed title bar:

![image](https://github.com/user-attachments/assets/548c31be-322d-46b8-adef-60ab98e58f9f)

It's a bit too large for my liking (rocking a KDE desktop where the title bars are a lot more compact) —w—

In theory I can set `gtk-titlebar = false`, but that reverts to the Breeze title bar, which does not respect `window-theme = ghostty`.

![image](https://github.com/user-attachments/assets/641bed2e-f088-4c3a-8929-0a7705928f3a)

I'd really like a small title bar like this, but with my own custom colors.

**Comment by tomektom** [comment]
You can also change KDE color theme (or create own, it's pretty simple) to match Ghostty colors. Maybe this theme will match to Ghostty?
https://store.kde.org/p/1312310

**Reply by liquidev** [reply]
Thing is, I like Breeze and don't really want to change it! I'd just like ghostty's window to not clash with it.

**Reply by tomektom** [reply]
In this case I would change Ghostty theme to match my desktop :smile: 

But you can (I also didn't know until I checked for this) create window rule to change colors only for specific window class. As I did on screen – changed my beautiful and matching Gruvbox into ugly and nonmatching Breeze Light :smile: So find one color theme matching your Ghostty (or create own - change only colors responsible for titlebar in one of existing themes) :wink: 
![image](https://github.com/user-attachments/assets/925beb5c-4db6-4abf-a8f1-bf8e959a54cc)

**Reply by liquidev** [reply]
That sounds pretty sweet :smile: 

Still I think there'd be value in letting you disable the buttons and get a more compact title bar, I use GNOME on my laptop and the title bar is still a bit too much for me there.

**Comment by Gelmo** [comment]
> In theory I can set gtk-titlebar = false, but that reverts to the Breeze title bar, which does not respect window-theme = ghostty.

I've opened a separate discussion for this issue - https://github.com/ghostty-org/ghostty/discussions/3993

**Comment by 73** [comment]
I would like to suggest just a floating close button, like [Showtime](https://apps.gnome.org/Showtime):
![image](https://github.com/user-attachments/assets/a6f8e66d-3684-4818-a464-df8532115362)

This might correspond to `macos-titlebar-style = hidden`. I would love a feature like that.

**Reply by nbCloud91** [reply]
Yes, floating controls like this looks visually consistent! [BlackBox](https://flathub.org/apps/com.raggesilver.BlackBox) terminal emulator also has this feature.

**Reply by 73** [reply]
> Yes, floating controls like this looks visually consistent! [BlackBox](https://flathub.org/apps/com.raggesilver.BlackBox) terminal emulator also has this feature.

Not exactly. In BlackBox the whole title bar is hidden and slides in when the cursor reaches the top border of the window. 

I would like the controls always visible and the bar transparent. Like the MacOS feature.

**Comment by cmcpasserby** [comment]
@liquidev how did you get the thin kde title bar on there when i set gtk-titlebar to false i just get no title bar.

**Reply by Fjodor42** [reply, marked as answer]
This discussion seems to have been started before version 1.1.0 was released.

With that version, `gtk-titlebar = false`, as the name implies, gives a window with no title bar.

The solution to get the simple title bar is `window-decoration = server`, but as of the 1.1.0, this is not available on X11, only on Wayland.

`window-decoration = server` support for X11 **has been added** to the code by @pluiedev (thank you!), but since that code was merged after the release of 1.1.0, it will only be available once the next release happens - unless you want to compile the current code yourself, obviously.

For discussions of what I mention here, there are a large number of discussions mentioning title bar and SSD (Server-Side Decorations), which is what the option above sets, as opposed the the default of Cleint-Side Decorations, CSD.
